### PR TITLE
Tighten bolts on the armory

### PR DESCRIPTION
**Pull request by AlCutter**
- Adds support for distinct firmware image hash, and firmware measurement hashes
- update docs

### CODE REVIEW

File: binary_transparency/firmware/cmd/publisher/publish.go [hunk 1]
```diff
@@ -117,13 +122,28 @@ func main() {
...
 
 func createManifestFromFlags() (api.FirmwareMetadata, []byte, error) {
+	var measure func([]byte) ([]byte, error)
+	switch *deviceID {
+	case "armory":
+		measure = usbarmory.ExpectedMeasurement
+	case "dummy":
+		measure = dummy.ExpectedMeasurement
+	default:
+		glog.Exitf("--device must be one of: 'dummy', 'armory'")
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/publisher/publish.go:132]: This should return the error from here and `main` should Exit.

File: binary_transparency/firmware/devices/dummy/rom/rom.go [hunk 2]
```diff
@@ -65,13 +65,20 @@ func ResetFromFlags() (Chain, error) {
...
 
-	// TODO(al): measure firmware and check against manifest expected measurement.
-	// HACK: this should use the firmware measurement field, but it'll do for now.
 	fw, err := ioutil.ReadFile(fwFile)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read firmware: %w", err)
 	}
-	fwHash := sha512.Sum512(fw)
+
+	// measure firmware
```
Comment (mhutchinson) [inline binary_transparency/firmware/devices/dummy/rom/rom.go:73]: Can this just call out to `dummy.ExpectedMeasurement(fw)`?

**Review by mhutchinson — APPROVED** [review]

File: binary_transparency/firmware/cmd/publisher/publish.go (hunk 1 above)
Comment (AlCutter) [reply binary_transparency/firmware/cmd/publisher/publish.go]: done, ta

File: binary_transparency/firmware/devices/dummy/rom/rom.go (hunk 2 above)
Comment (AlCutter) [reply binary_transparency/firmware/devices/dummy/rom/rom.go]: Yeah, I had some half baked idea about keeping the generation and measurement of this separate, but your suggestion results in simpler code so I'll do that for now and break out later if need be.

**Pull request merged**